src/multi.c

```diff
@@ -97,15 +97,14 @@ bool MultiHandleCommand(RedisRaftCtx *rr,
          * intercepted and reject the transaction.
          */
         unsigned int cmd_flags = CommandSpecGetAggregateFlags(cmds, 0);
-        raft_term_t term = raft_get_current_term(rr->raft);
 
         if (cmd_flags & CMD_SPEC_UNSUPPORTED) {
             RedisModule_ReplyWithError(ctx, "ERR not supported by RedisRaft");
             multiState->error = true;
         } else if (cmd_flags & CMD_SPEC_DONT_INTERCEPT) {
             RedisModule_ReplyWithError(ctx, "ERR not supported by RedisRaft inside MULTI/EXEC");
             multiState->error = true;
-        } else if (cmd_flags & CMD_SPEC_DENYOOM && (RedisModule_GetUsedMemoryRatio() > 1.0 || term != rr->snapshot_info.last_applied_term)) {
+        } else if (RedisModule_GetUsedMemoryRatio() > 1.0) {
```

Comment:
I don't think this change is correct.  If one has a multi of all GETs, it should work just fime.  i.e. the cmd-flags & CMD_SPEC_DENYOOM is needed.

Reply:
I mentioned in the comment above, this is how Redis works right now. I don't agree much with this behavior but I think we should try to follow Redis. This is why I've made this change. 

Please see: https://github.com/redis/redis/issues/9926. Someone complains about it (rightfully) and it is still unresolved. If Redis solves it, we can follow it. Wdyt?

Reply:
ok, I see.  I was confused by the special work done for scripts.  as noted in the thread there seems inconsistent to treat eval and multi differently, but <shrug>